oci8_sql_number_go112_test.go

```diff
@@ -0,0 +1,157 @@
+// +build go1.12
+
+package oci8
+
+import (
+	"context"
+	"database/sql"
+	"testing"
+)
+
+// TestDestructiveNumberCursor checks select cursor
+func TestDestructiveNumberCursor(t *testing.T) {
+	if TestDisableDatabase || TestDisableDestructive {
+		t.SkipNow()
+	}
+
+	tableName := "number_cursor_" + TestTimeString
+	err := testExec(t, "create table "+tableName+" ( A INTEGER )", nil)
+	if err != nil {
+		t.Fatal("create table error:", err)
+	}
+
+	defer testDropTable(t, tableName)
+
+	ctx, cancel := context.WithTimeout(context.Background(), TestContextTimeout)
+	stmt, err := TestDB.PrepareContext(ctx, "insert into "+tableName+" ( A ) values (:1)")
+	cancel()
+	if err != nil {
+		t.Fatal("prepare error:", err)
+	}
+
+	ctx, cancel = context.WithTimeout(context.Background(), TestContextTimeout)
+	_, err = stmt.ExecContext(ctx, 1)
+	cancel()
+	if err != nil {
+		stmt.Close()
+		t.Fatal("exec error:", err)
+	}
+
+	ctx, cancel = context.WithTimeout(context.Background(), TestContextTimeout)
+	_, err = stmt.ExecContext(ctx, 2)
+	cancel()
+	if err != nil {
+		stmt.Close()
+		t.Fatal("exec error:", err)
```

Comment:
Thanks for pull-requset. For release resources with safety, how about to make private function testDestructiveNumberCursor that use defer and return and cal it from TestDestructiveNumberCursor?

Reply:
It is for a test, even if there was a mistake with resource release, it is not going to make much difference for testing.

I would rather have it split out to make sure resource close/cancel are called, this is how I caught the row cancel/close bug from last PR.

Reply:
Yes, I know. It is not a big deal for quality of go-oci8 with fixing this or not. But it makes sense if you want to run clean tests. :)

Reply:
Do you still want this changed before merging?

Reply:
Working on changing this